### DISCUSSION TITLE
Different "conn" sections in ipsec.conf with different encryption does`not work.

### DISCUSSION THREAD

**Discussion by TobRein** [strongswan/strongswan#596] [General]
Hello community,

I`am using strongswan 5.9.3.
I my ipesc.conf there are two "conn" section.
the first for a site-to-site connection with the follwoing encryption IKE_SA and Child_SA.
	ike=aes256gcm128-sha512-ecp512bp!
	esp=aes256gcm128-sha512-ecp512bp!
the first connection works

The second conn is for a road warrior more precisely for a android handy with the strongswan android client from the google play store.
The encryption is set to
	ike=aes128-sha256-modp2048!
	esp=aes128-sha256-modp2048!
But the strongswan android client app says :
	selected proposal: IKE: AES_GCM_16_256/PRF_HMAC_SHA_512/ECP_512_BP
	ECDH shared secret computation failed.

It seems that the second conn uses the encryption settings from the first conn ?

Can I use different IKE and ESP config entrys in the different conn section or can I use only one for all conn`s ?

Best regards
T. Reineck

**Comment by Thermi** [comment]
#196

**Comment by tobiasbrunner** [comment]
> selected proposal: IKE: AES_GCM_16_256/PRF_HMAC_SHA_512/ECP_512_BP
> ECDH shared secret computation failed.

At least in theory, the Android client should support the Brainpool curves (although the BoringSSL version we use doesn't natively support them, the _openssl_ plugin adds them manually, or at least it tries). I've not tried that in a long time, so maybe some of the recent changes in the plugin (e.g. those merged with 8e367df6db4cb244fa6f769bfd9c0d05a486b1c4) broke that support somehow.

> It seems that the second conn uses the encryption settings from the first conn ?

IKE configs are selected based on the IP addresses, IKE version and the received proposals, afterwards it's basically the order in the config that determines what's selected. So consider that when writing your config (it might require adding additional algorithms to the first config's IKE proposal - note that there could be a config switch after based on the identities so the ESP proposal might not need that).

**Comment by TobRein** [comment]
Hello tobiasbrunner,

thank you for your answer.
The inital problem was that both "conn" right entries are set to %any so I think thats why 
the second conn does not get the correct esp and ike values.
I set the first conn right parameter to a IP range with the encryiption of
	ike=aes256gcm128-sha512-ecp512bp!
	esp=aes256gcm128-sha512-ecp512bp!
and it works fine.

The second conn right parameter stays on %any.
With the follwing ike and esp parameters
	ike=aes256gcm128-sha512-modp8192!
	esp=aes256gcm128-sha512-modp8192!
and it works.
The "ecp512bp" does not work on my mobile phone.
Anyway with the "aes256gcm128-sha512-modp8192" parameters I can live with that.

Best regrads
T. Reineck
